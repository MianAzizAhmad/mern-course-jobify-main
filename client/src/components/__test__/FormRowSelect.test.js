import React from 'react';
import FormRowSelect from './../FormRowSelect';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

describe('Test Case { ', ()=>{

    it('FormRowSelect Snap 1 }', ()=>{
        const objects = [{itemValue : "Value1", index : "1"}, {itemValue : "Value2", index : "2"}, {itemValue : "Value3", index : "3"}]
        const tree = renderer.create(<FormRowSelect name="name 1" value="test 1" handleChange="No change" labelText="Label1" list={objects} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('FormRowSelect Snap 2 }', ()=>{
        const tree = renderer.create(<FormRowSelect ></FormRowSelect>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('FormRowSelect Snap 3 }', ()=>{
        const objects = [{itemValue : "", index : "1"}, {itemValue : "Value2", index : ""}, {itemValue : "Value3", index : "3"}]
        const tree = renderer.create(<FormRowSelect name="snap3" value="SNAP 3" handleChange="Change" labelText="Password" list={objects}></FormRowSelect>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});