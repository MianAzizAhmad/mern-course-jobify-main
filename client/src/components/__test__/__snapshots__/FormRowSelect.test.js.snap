// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case {  FormRowSelect Snap 1 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
    htmlFor="name 1"
  >
    Label1
  </label>
  <select
    className="form-select"
    name="name 1"
    onChange="No change"
    value="test 1"
  >
    <option
      value="Value1"
    >
      Value1
    </option>
    <option
      value="Value2"
    >
      Value2
    </option>
    <option
      value="Value3"
    >
      Value3
    </option>
  </select>
</div>
`;

exports[`Test Case {  FormRowSelect Snap 2 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
  />
  <select
    className="form-select"
  />
</div>
`;

exports[`Test Case {  FormRowSelect Snap 3 } 1`] = `
<div
  className="form-row"
>
  <label
    className="form-label"
    htmlFor="snap3"
  >
    Password
  </label>
  <select
    className="form-select"
    name="snap3"
    onChange="Change"
    value="SNAP 3"
  >
    <option
      value=""
    />
    <option
      value="Value2"
    >
      Value2
    </option>
    <option
      value="Value3"
    >
      Value3
    </option>
  </select>
</div>
`;
